packages/starlight/__tests__/basics/base.test.ts

```diff
@@ -0,0 +1,37 @@
+import { describe, expect, test, vi } from 'vitest';
+import { fileWithBase, pathWithBase } from '../../utils/base';
+
+describe('fileWithBase()', () => {
+  describe('with no base', () => {
+    test('does not prepend anything', () => {
+      expect(fileWithBase('/img.svg')).toBe('/img.svg');
+    });
+    test('adds leading slash if needed', () => {
+      expect(fileWithBase('img.svg')).toBe('/img.svg');
+    });
+  });
+
+  // TODO: Stubbing BASE_URL is not currently possible.
+  // Astro controls BASE_URL via its `vite-plugin-env`, which prevents Vitest’s stubbing from
+  // working and there’s also no way to pass in Astro config in Astro’s `getViteConfig` helper.
+  describe.todo('with base', () => {
+    test('prepends base', () => {
+      vi.stubEnv('BASE_URL', '/base/');
+      expect(fileWithBase('/img.svg')).toBe('/base/img.svg');
+      vi.unstubAllEnvs();
+    });
+  });
```

Comment:
Vitest provides a [stubbing helper](https://vitest.dev/api/vi.html#vi-stubenv) to set `import.meta.env` values in tests, but this does not work for values controlled by Astro’s [`vite-plugin-env`](https://github.com/withastro/astro/blob/e80896a67c08c1c7916417d2f385a185c773d97e/packages/astro/src/vite-plugin-env/index.ts) such as `BASE_URL`.

It is also not possible to set Astro’s `base` option via a `vitest.config.ts` file using Astro’s `getViteConfig` helper — instead Astro side loads an `astro.config` from disk. I couldn’t find a way to get it to detect a local test-suite-specific `astro.config`, so for now testing for custom `base` settings seems to be blocked.

Reply:
Yeah, that would be the hardcore way (although given https://github.com/withastro/astro/issues/7561 might also be tricky to set-up without cluttering the repo with fixture directories at the base level), but seems wrong for testing a single function output 😅